front/components/assistant/AssistantDetails.tsx

```diff
@@ -287,23 +289,52 @@ export function AssistantDetails({
           {otherActions.map((action, index) =>
             isDustAppRunConfiguration(action) ? (
               <div className="flex flex-col gap-2" key={`action-${index}`}>
-                <div className="text-lg font-bold text-element-800">Action</div>
+                <div className="text-lg font-bold text-element-800">
+                  Run Actions
+                </div>
                 <DustAppSection dustApp={action} owner={owner} />
               </div>
             ) : isTablesQueryConfiguration(action) ? (
               <div className="flex flex-col gap-2" key={`action-${index}`}>
-                <div className="text-lg font-bold text-element-800">Tables</div>
-                <TablesQuerySection tablesQueryConfig={action} />
+                <div className="text-lg font-bold text-element-800">
+                  Query Tables
+                </div>
+                <DataSourceViewsSection
+                  owner={owner}
+                  dataSourceViews={dataSourceViews}
+                  dataSourceConfigurations={action.tables.map((t) => {
+                    // We should never have an undefined dataSourceView here as
+                    // if it's undefined, it means the dataSourceView was deleted and the configuration is invalid
+                    // But we need to handle this case to avoid crashing the UI
+                    const dataSourceView = dataSourceViews.find(
```

Comment:
We should not have to fake this. IIRC, the `filter` is used downstream only to check if all nodes have been selected. Can't we adapt the component so it handles either data source or table?

Reply:
> the filter is used downstream only to check if all nodes have been selected

Not only it seems, it also affect the view (by adding "in: [t.tableId]" I could have the same display as in the datasource selector.).

Filter could be made optional and both types would be accepted but I prefer to have a coherent UI.

WDYT ?

Reply:
I feel the component should take care of this for you then, a bit like what `DataSourceSelectionSection` does actually. Where we have a common type for both documents and tables.

Reply:
So I went down that road and it was making it even more complicated.

My take is to merge as it and revisit together to try use a common type (like in AssistantBuilder).

Reply:
Then add a `// TODO(GROUPS_INFRA):` 🙏